share/disambiguation.py

```diff
@@ -125,24 +122,22 @@ def _instance_for_node(self, node):
             constrain.append(Q(type__in=node.model.get_types()))
             constrain.append(Q(type=node.model._typedmodels_type))
 
+        # HACK
+        unmerged_query = Q(same_as__isnull=True) if hasattr(concrete_model, 'same_as') else Q()
+
         for q in constrain:
```

Comment:
What if we use the existing `constrain_types` logic for agents, instead of this `constrain` list?
Disambiguating an `institution` would match `agent`, `organization`, or `institution`, so it'll merge in any generic agents with the same name. Disambiguating an `organization` would match `agent`, `organization`, `institution`, `consortium`, or `department`, which could potentially merge things that shouldn't be, but only with the same name/identifier/cited_as, and never people.